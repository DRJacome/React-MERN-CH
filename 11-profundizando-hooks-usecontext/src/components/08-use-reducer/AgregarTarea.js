import React from 'react'
import { useForm } from "../../hooks/useForm"

export const AgregarTarea = ({ gestionarAgregarTarea }) => {
  /* const [{valoresFormulario}, gestionarInput] = useForm({ */
  const [{ descripcion }, gestionarInput, reset] = useForm({
    descripcion: ''
  })

  const gestionarFormulario = (evento) => {
    evento.preventDefault();
    /* Si no se escribe nada en el campo input, 
    el programa no hará nada (no añadirá una nueva tarea) */
    if (descripcion.trim().length <= 1) {
      return null;
    }

    const nuevaTarea = {
      id: new Date().getTime(),
      descripcion: descripcion,
      hecho: false
    }

    gestionarAgregarTarea(nuevaTarea)
    reset()
  }

  return (
    <>
      <h4>Agregar tarea</h4>
      <hr />
      <form onSubmit={gestionarFormulario}>
        <input
          type='text'
          name='descripcion'
          className='form-control'
          placeholder='Aprender...'
          autoComplete='off'
          onChange={gestionarInput}
          value={descripcion}
        />
        <button
          className='btn btn-outline-primary mt-1 btn-block'
          type='submit'
        >Agregar
        </button>
      </form>
    </>
  )
}