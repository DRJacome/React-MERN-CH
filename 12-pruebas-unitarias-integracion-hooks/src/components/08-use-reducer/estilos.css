body {
  padding: 50px;
  background: lightgrey;
}

p {
  cursor: pointer;
  margin: 0px;
}

.list-group-item {
  align-items: center;
  display: flex;
  justify-content: space-between;
}

.completada{
  text-decoration: line-through;
}