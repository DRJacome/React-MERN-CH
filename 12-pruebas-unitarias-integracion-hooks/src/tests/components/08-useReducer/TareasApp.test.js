import React from 'react'
import { mount, shallow } from 'enzyme'
import { TareasApp } from '../../../components/08-use-reducer/TareasApp';
import { demoTareas } from '../../fixtures/demoTareas';
import { act } from '@testing-library/react'

describe('Pruebas en <TareasApp/>', () => {

  const wrapper = shallow(<TareasApp />);
  Storage.prototype.setItem = jest.fn(() => { })

  test('Debe renderizar el componente correctamente', () => {

    expect(wrapper).toMatchSnapshot();
  })

  test('Debe agregar un componente <ListaTareas/>', () => {

    const wrapper = mount(<TareasApp />)

    act(() => {
      wrapper.find('AgregarTarea').prop('gestionarAgregarTarea')(demoTareas[0]);
      wrapper.find('AgregarTarea').prop('gestionarAgregarTarea')(demoTareas[1]);
    })

    expect(wrapper.find('h1').text().trim()).toBe('TareasApp (2)');
    expect(localStorage.setItem).toHaveBeenCalledTimes(2);
  })

  test('Debe eliminar un <ListaTareas/>', () => {

    wrapper.find('AgregarTarea').prop('gestionarAgregarTarea')(demoTareas[0])
    expect(wrapper.find('h1').text().trim()).toBe('TareasApp (1)');

    wrapper.find('ListaTareas').prop('gestionarBorrado')(demoTareas[0].id)

    expect(wrapper.find('h1').text().trim()).toBe('TareasApp (0)');
  })
})
