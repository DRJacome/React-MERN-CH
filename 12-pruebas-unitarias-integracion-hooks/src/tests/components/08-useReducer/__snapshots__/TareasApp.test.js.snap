// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TareasApp/> Debe renderizar el componente correctamente 1`] = `
<div>
  <h1>
    TareasApp (
    0
    )
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <ListaTareas
        gestionarBorrado={[Function]}
        gestionarTachado={[Function]}
        tareas={Array []}
      />
    </div>
    <div
      className="col-5"
    >
      <AgregarTarea
        gestionarAgregarTarea={[Function]}
      />
    </div>
  </div>
</div>
`;
